<a href="https://colab.research.google.com/github/seok-jin/kaggle_study/blob/main/Santander_product_recomendation.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# 1.탐색적 데이터 분석 

In [2]:
import os
import pandas as pd
import numpy as np
os.chdir('/content/drive/MyDrive/Colab Notebooks')
trn  = pd.read_csv('train_ver2.csv')

/usr/local/lib/python3.7/dist-packages/IPython/core/interactiveshell.py:2718: DtypeWarning: Columns (5,8,11,15) have mixed types.Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


## 1.1기초 통계로 데이터 살펴보기
- .shape 함수를 사용하여 데이터의 크기를 확인

In [3]:
trn.shape 

(13647309, 48)

- .head 함수를 통해 데이터의 첫 5줄을 확인

In [ ]:
trn.head()

fecha_dato  ncodpers  ... ind_nom_pens_ult1 ind_recibo_ult1
0  2015-01-28   1375586  ...               0.0               0
1  2015-01-28   1050611  ...               0.0               0
2  2015-01-28   1050612  ...               0.0               0
3  2015-01-28   1050613  ...               0.0               0
4  2015-01-28   1050614  ...               0.0               0

[5 rows x 48 columns]

-  for loop 문으로 모든 변수의 첫 5줄 미리보기 

  - 컬럼별로 데이터 타입을 확인 할 수 있기 때문에 유용하다.



In [ ]:
for col in trn.columns:
  print('{}\n'.format(trn[col].head()))

0    2015-01-28
1    2015-01-28
2    2015-01-28
3    2015-01-28
4    2015-01-28
Name: fecha_dato, dtype: object

0    1375586
1    1050611
2    1050612
3    1050613
4    1050614
Name: ncodpers, dtype: int64

0    N
1    N
2    N
3    N
4    N
Name: ind_empleado, dtype: object

0    ES
1    ES
2    ES
3    ES
4    ES
Name: pais_residencia, dtype: object

0    H
1    V
2    V
3    H
4    V
Name: sexo, dtype: object

0     35
1     23
2     23
3     22
4     23
Name: age, dtype: object

0    2015-01-12
1    2012-08-10
2    2012-08-10
3    2012-08-10
4    2012-08-10
Name: fecha_alta, dtype: object

0    0.0
1    0.0
2    0.0
3    0.0
4    0.0
Name: ind_nuevo, dtype: float64

0          6
1         35
2         35
3         35
4         35
Name: antiguedad, dtype: object

0    1.0
1    1.0
2    1.0
3    1.0
4    1.0
Name: indrel, dtype: float64

0    NaN
1    NaN
2    NaN
3    NaN
4    NaN
Name: ult_fec_cli_1t, dtype: object

0    1
1    1
2    1
3    1
4    1
Name: indrel_1mes, dtype: obje

- pandas info()를 활용하여 더욱 자세하게 분석 가능하다.

In [ ]:
trn.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13647309 entries, 0 to 13647308
Data columns (total 48 columns):
 #   Column                 Dtype  
---  ------                 -----  
 0   fecha_dato             object 
 1   ncodpers               int64  
 2   ind_empleado           object 
 3   pais_residencia        object 
 4   sexo                   object 
 5   age                    object 
 6   fecha_alta             object 
 7   ind_nuevo              float64
 8   antiguedad             object 
 9   indrel                 float64
 10  ult_fec_cli_1t         object 
 11  indrel_1mes            object 
 12  tiprel_1mes            object 
 13  indresi                object 
 14  indext                 object 
 15  conyuemp               object 
 16  canal_entrada          object 
 17  indfall                object 
 18  tipodom                float64
 19  cod_prov               float64
 20  nomprov                object 
 21  ind_actividad_cliente  float64
 22  renta           

## 데이터 크기 
- RangeIndex, Data columns 값은 데이터의 열과 행의 크기를 나타낸다.
- shape 함수를 활용하여 확인 가능하다.(13,647,309개의 고객 데이터, 48개의 변수 데이터)

In [6]:
num_cols = [col for col in trn[:24] if trn[col].dtype in ['int64', 'float64']]

In [1]:
trn[num_cols].describe()

NameError: ignored